### PR TITLE
Use owner.factoryFor is present

### PR DESCRIPTION
**Pull request by cibernox**
This removes deprecations and should have a very positive perf outcome in Ember 2.12

### CODE REVIEW

**Comment by workmanw** [comment]
Just looking at these changes, it's not obvious to me where the performance improvement comes from. I was wondering if you could elaborate on that or maybe point me toward something I could read.

**Comment by cibernox** [comment]
The factoryFor method seems to be way faster, according with the release notes of 2.12.
And even if it wasn't, the other method is deprecated.

**Comment by rwjblue** [comment]
The factory for RFC outlines the perf benefits. The short version is that Ember no longer has to `.extend` every factory a second time (which we have done for a long time to ensure it gets the proper injections).

**Comment by workmanw** [comment]
Oh awesome! Thank you @cibernox @rwjblue. I read the PR, but it wasn't clear to me there. I will definitely go reread the RFC. :+1:

**Comment by cibernox** [comment]
@bwbuchanan Added

File: addon/utils/locale.js [hunk 1]
```diff
@@ -115,19 +133,36 @@ function getFlattenedTranslations(id, owner) {
     assign(result, getFlattenedTranslations(parentId, owner));
   }
 
-  const translations = owner._lookupFactory(`locale:${id}/translations`) || {};
-  assign(result, withFlattenedKeys(translations));
+  let translations;
+  if (owner.factoryFor) {
+    let maybeTranslations = owner.factoryFor(`locale:${id}/translations`);
+    translations = translations && translations.class;
```
Comment (bwbuchanan) [inline addon/utils/locale.js]: Should be `translations = maybeTranslations && maybeTranslations.class;`

**Comment by bwbuchanan** [comment]
Looks good. My app's tests are green.

**Comment by cibernox** [comment]
Actually mine had so many deprecations that tests broke because travis cannot handle a log that big.

File: addon/services/i18n.js [hunk 2]
```diff
@@ -69,7 +69,13 @@ export default Parent.extend(Evented, {
 
   // @private
   _initDefaults: on('init', function() {
-    const ENV = getOwner(this)._lookupFactory('config:environment');
+    let owner = getOwner(this);
+    let ENV;
+    if (owner.factoryFor) {
+      ENV = owner.factoryFor('config:environment').class;
```
Comment (rwjblue) [inline addon/services/i18n.js]: In general, we should return the object returned by factoryFor (not the .class property)

File: addon/utils/add-translations.js [hunk 3]
```diff
@@ -4,7 +4,13 @@ const assign = Ember.assign || Ember.merge;
 
 export default function addTranslations(locale, newTranslations, owner) {
   const key = `locale:${locale}/translations`;
-  var existingTranslations = owner._lookupFactory(key);
+  let existingTranslations;
+  if (owner.factoryFor) {
+    let maybeTranslations = owner.factoryFor(key);
+    existingTranslations = maybeTranslations && maybeTranslations.class;
```
Comment (rwjblue) [inline addon/utils/add-translations.js]: Remove .class

**Comment by cibernox** [comment]
**Don't merge** this yet. I've realized that I misunderstood how `factoryFor` is supposed to be used. I'll update this PR.

**Pull request merged**